inventory_management_system_api/migrations/core.py

```diff
@@ -0,0 +1,223 @@
+    migrations_collection.update_one({"_id": "previous_migration"}, {"$set": {"name": name}}, upsert=True)
+
+
+def find_migration_index(name: str, migration_names: list[str]) -> int:
+    """
+    Returns the index of a specific migration name in a list of sorted migration names.
+
+    :param name: Name of the migration to look for. A value of 'latest' indicates the last available one should be used
+                 instead.
+    :param migration_names: List of migration names.
+    :return: Index of the found migration in the `migration_names` list.
+    :raises: ValueError if the `name` does not appear in `migration_names`.
+    """
+
+    if name == "latest":
+        return len(migration_names) - 1
+    return migration_names.index(name)
+
+
+def load_migrations_forward_to(name: str) -> dict[str, BaseMigration]:
+    """
+    Returns a list of migrations forward that need to be applied to get from the last migration applied to the database
+    to the given one inclusive.
+
+    :param name: Name of the last migration forward to apply. 'latest' will just use the latest one.
+    :returns: List of dicts containing the names and instances of the migrations that need to be applied in the order
+              they should be applied.
+    """
+
+    available_migrations = find_available_migrations()
+
+    start_index = 0
+
+    previous_migration = get_previous_migration()
+    if previous_migration:
+        try:
+            start_index = find_migration_index(previous_migration, available_migrations) + 1
+        except ValueError:
+            logger.warning(
+                "Previous migration applied '%s' not found in current migrations. Have you skipped a version?",
+                previous_migration,
+            )
+
+    try:
+        end_index = find_migration_index(name, available_migrations)
+    except ValueError:
+        sys.exit(f"Migration '{name}' was not found in the available list of migrations.")
+
+    if start_index > end_index:
+        sys.exit(
+            f"Migration '{name}' is before the previous migration applied '{previous_migration}'. So there is nothing "
+            "to migrate."
+        )
+
+    # Dicts are insertion ordered so will match the list order
+    return {name: load_migration(name) for name in available_migrations[start_index : end_index + 1]}
+
+
+def load_migrations_backward_to(name: str) -> tuple[dict[str, BaseMigration], Optional[str]]:
+    """
+    Returns a list of migrations backward that need to be applied to get from the last migration applied to the database
+    to the given one inclusive.
+
+    :param name: Name of the last migration backward to apply.
+    :returns: Tuple containing:
+              - List of dicts containing the names and instances of the migrations that need to be applied in the order
+                they should be applied.
+              - Either the name of the last migration before the one given or `None` if there aren't any.
+    """
```

Comment:
The backwards migration reverses the specified migration (since it is inclusive). Arguably it could be better set to exclusive instead, but it depends on what use cases you imagine it would be used for? Or if you just wanted to maintain consistency with the forwards migration, that would be reasonable as well.

Reply:
Yeah I had the same thought, the reason I didn't in the end is to keep consistency with alembic as it downgrades inclusive of the revision as this PR is effectively recreating it for mongodb. Arguably I could even replace the command names so that the current `forward` -> `upgrade`, `backward` -> `downgrade`, `set` -> `stamp`, `latest` -> `head`. But I left it with what we were already using for the forward and backward, and head doesn't make as much sense here as its not quite fully the same.